guides/20240823_how_to_setup_postgresql_playground_in_daytona.md

```diff
@@ -0,0 +1,372 @@
+
+## Creating a DevContainer for PostgreSQL
+
+A development container (or a devcontainer for short) allows you to use a docker container as a full-featured development environment. It can be configured to meet your development environment needs. It could include tools and runtimes like npm, git, maven, Golang compiler and others too.
+
+For this guide we're going to create a devcontainer for PostgreSQL using a config file. The file is always named `devcontainer.json` with code syntax in it following the correct config specifications. Its a norm to keep the file in an hidden directory call `.devcontainer`
+
+Let's get started. I will use the terminal to create mine in my linux PC but the commands should work fine on Mac terminal or the Windows powershell.
+
+- ### **Step 1**: Create a Directory
+
+  Create a directory with any name of your choice and go into it. I use the name `postgresql-playground-in-daytona` and move into the directory.
+
+  ```bash
+  mkdir postgresql-playground-in-daytona && cd postgresql-playground-in-daytona
+  ```
+
+- ### **Step 2**: Create the `.devcontainer` directory
+
+  Create a hidden directory called `.devcontainer` and enter it. This is where our devcontainer config file will be stored.
+
+  ```bash
+  mkdir .devcontainer && cd .devcontainer
+  ```
+
+- ### **Step 3**: Create `devcontainer.json` file
+
+  Now, create a file called `devcontainer.json` and paste the following code into it and then save it.
+
+  ```json
+  {
+      "name": "PostgreSQL Dev Container Playground",
+      "dockerComposeFile": "docker-compose.yml",
+      "workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+      "service": "postgres",
+      "forwardPorts": [5432],
+      "customizations": {
+        "vscode": {
+            "extensions": ["streetsidesoftware.code-spell-checker"]
+        }
+      },
+      "postCreateCommand": "psql -U ${POSTGRES_USER} -c 'SELECT 1;'"
+  }
+  ```
+
+  The `devcontainer.json` code content defines a configuration for a PostgreSQL development container environment using Docker Compose
+
+  - **`name`:** Set the name of the development container environment to ``PostgreSQL Dev Container Playground``.
+  - **`dockerComposeFile`:** Specifies the Docker Compose file (`docker-compose.yml`) that defines the services and their configurations.
+  - **`workspaceFolder`:** Determine the location of the workspace folder within the container, using the base name of the local workspace folder.
+  - **`service`:** Indicate that the development container will primarily interact with the ``postgres`` service defined in the Docker Compose file.
+  - **`forwardPorts`:** Forwards the container's port ``5432`` to the host machine, allowing local access to the PostgreSQL database.
+  - **`customizations`:** Defines additional customizations for the development environment:
+      - **`vscode`:** Specifies customizations for Visual Studio Code:
```

Comment:
please do not use lists with several levels, this is not supported in our cms, only 1st level

Reply:
I had to remove it since I made use of a feature managed by someone as advised by you